### PR TITLE
Space dragons can now catch and eat fish (plus shark form and rift fishing)

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This started as a yet one of those stupid jokes about fishing in coderbus, until I was hooked on the idea of space dragons being able to eat fish, which also lead me to embark on a two-days long journey of spriting and then coding the a shark form for space dragons, the achievement for unlocking it, and a baby space carp fish for the carp rift.

The shark form is a basically reward for eating a ton of fish, like a lot. You'll probably need time and help to do it, which more often than not means coming to a truce with the station for the sake of a gimmick with some mild effects on the way space dragons regenerate health (more from carps and fish, less from other mobs) plus some fishing skill.

Luckily, this only has to be done once. Once the achievement is earned, joining as a space dragon will ask you if you want to switch to shark form, unless you're taking over someone else's dragon which has been already named and colored. Otherwise, having to do it every round would make the feature boring, tedious and ephemeral.

Oh, did I mention that space carps and space dragons can now fish (like lobstrosities)? that the space dragon is better at it than carps? That the carp rift is a fishing spot now, with a carp-related loot table? That baby carps can be placed into an aquarium and they'll grow into actual space carps in half a dozen minutes? Well, now I have.

By the by, screenshot for how space shark-dragons look in game:
immagine

And the soulless animation loop for the space shark-dragon that took me a while to sprite (yes, I didn't do more dirs, lazy me ig).
spacecarp_anim-export

## Why It's Good For The Game
I like the idea of space dragons being able to eat fish. They're pretty much carnivorous apex predators. The rest is stuff I built around the assertion that space carps and dragons can now fish. Moreover, I think it's ok to have achievements that actually reward you with something beside the cool cheevo itself, though the reward in this case is something with mild gameplay effects, though I would hardly call it a buff. May be subject to change.

## Changelog

:cl:
add: Space carps and space dragons can now fish. Space dragons can now eat fish too, for a very mild healing.
add: The carp rift is now a fishing spot, filled with carp-related loot, including baby space carps (aquarium fish that eventually grows into a real space carp in a matter of half a dozen minutes).
add: Added a shark form (and relative achievement) to space dragons, unlocked after eating an unspecificatedly high amount of fish.
/:cl:

### CODE REVIEW

**Comment by Hotmud9** [comment]
I'd say sharks could be a new addition to space dragons; something similar to an alien queen's praetorian
Could give space dragons some spice

**Comment by TheSmallBlue** [comment]
Big blahaj.....

**Comment by the-orange-cow** [comment]
Catches-The-Fish:  Don't kill the dragon yet!  I'm not done!

**Comment by Xkallubar** [comment]
Sprite has too many pixels to fit in tbh

**Comment by Ghommie** [comment]
> Sprite has too many pixels to fit in tbh

Perhaps it's a big chonky, but it's about the same as the normal space dragon in terms of height, a bit narrower on the width. Shockingly, we also have mobs that are even bigger than it. They're called megafauna.

**Comment by moocowswag** [comment]
This is going to generate so much drama with valid hunters going for a space dragon and people who are just chilling fishing from a carp rift.

File: code/modules/mob/living/basic/space_fauna/space_dragon/space_dragon.dm [hunk 1]
```diff
@@ -89,9 +95,31 @@
...
+	. = ..()
+	if(shark_form && mind.get_skill_level(/datum/skill/fishing) < SKILL_LEVEL_APPRENTICE)
+		mind.set_level(/datum/skill/fishing, SKILL_LEVEL_APPRENTICE, TRUE)
+
+/mob/living/basic/space_dragon/proc/sharkify()
+	pixel_z -= 3
+	base_pixel_z -= 3
+	do_jitter_animation(150)
+	shark_form = TRUE
+	desc = "A piscine mutation of the fearsome leviathan whose flight defies modern physics. Said to be the other ultimate stage in the lige cycle of the Space Carp."
```
Comment (norsvenska) [inline code/modules/mob/living/basic/space_fauna/space_dragon/space_dragon.dm:114]: lige

**Comment by Admiral-Gigaton** [comment]
If this gets merged with https://github.com/tgstation/tgstation/pull/86291#issue-2499168690

Then the arguement of carp rifts disappearing MIGHT be able to be shut down if the portal can stay linked to the rift loot pool, AFTER the rift is killed.

Or walk to a rift that is fully open while dragon is making a 2nd or 3rd to avoid getting mauled to death

**Comment by Ghommie** [comment]
Oh, I've just realized there's a lot of ambiguousity around what rift I'm talking about.

With "carp rift", I'm referring to **the rift that space dragons place and from which space carps are regularly spawned**, not the one that carps use to teleport around. However I'll probably add something to regulard rifts in the future too.

**Review by Wallemations — APPROVED** [review]
I approve big shark

**Comment by Dawnseer** [comment]
> If this gets merged with [#86291 (comment)](https://github.com/tgstation/tgstation/pull/86291#issue-2499168690)
> 
> Then the arguement of carp rifts disappearing MIGHT be able to be shut down if the portal can stay linked to the rift loot pool, AFTER the rift is killed.
> 
> Or walk to a rift that is fully open while dragon is making a 2nd or 3rd to avoid getting mauled to death.

Of course, this should let carp spawn from the fishing portal if it's linked to the carp rift. Local fisherman accidentally causing the stations destruction.

**Comment by Admiral-Gigaton** [comment]
> > If this gets merged with [#86291 (comment)](https://github.com/tgstation/tgstation/pull/86291#issue-2499168690)
> > Then the arguement of carp rifts disappearing MIGHT be able to be shut down if the portal can stay linked to the rift loot pool, AFTER the rift is killed.
> > Or walk to a rift that is fully open while dragon is making a 2nd or 3rd to avoid getting mauled to death.
> 
> Of course, this should let carp spawn from the fishing portal if it's linked to the carp rift. Local fisherman accidentally causing the stations destruction.

emag function #435 spotted, *FETCH ME THAT* ***BALANCED INTERACTION***

**Pull request merged**